rs_client/stac_client.py

```diff
@@ -328,3 +330,72 @@ def remove_item(  # type: ignore # pylint: disable=arguments-differ
             **self.apikey_headers,
             timeout=timeout,
         )
+
+    DatetimeOrTimestamp = Optional[Union[datetime_, str]]
```

Comment:
Rather than copy/pasting them from pystac, just use something like: 
```python

from pystac_client.item_search import (
    BBoxLike,
    CollectionsLike,
    DatetimeLike,
    FieldsLike,
    FilterLangLike,
    FilterLike,
    IDsLike,
    IntersectsLike,
    ItemSearch,
    QueryLike,
    SortbyLike,
)
```